src/main/scala/beam/sim/common/Range.scala

```diff
@@ -26,33 +26,58 @@ case class Range(lowerBound: Int, upperBound: Int) {
 
 object Range {
 
-  def apply(pattern: String): Range =
-    if (pattern == null || pattern.isEmpty) Range.empty()
+  val pattern = """\[-?\d*\:-?\d*\]"""
```

Comment:
I think this will break some scenarios. For example, I the data for VehicleEnergy is in the format (#,#].

Reply:
it uses in specific case.